internal/services/logpull_retention/resource_test.go

```diff
@@ -19,24 +22,48 @@ func TestAccLogpullRetentionSetStatus(t *testing.T) {
 	}
 
 	rnd := utils.GenerateRandomResourceName()
-	name := "cloudflare_logpull_retention." + rnd
+	resourceName := "cloudflare_logpull_retention." + rnd
 	zoneID := os.Getenv("CLOUDFLARE_ZONE_ID")
 
 	resource.Test(t, resource.TestCase{
 		PreCheck:                 func() { acctest.TestAccPreCheck(t) },
 		ProtoV6ProviderFactories: acctest.TestAccProtoV6ProviderFactories,
 		Steps: []resource.TestStep{
+			// Set flag to false.
 			{
-				Config: testLogpullRetentionSetConfig(rnd, zoneID, "false"),
-				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr(name, consts.ZoneIDSchemaKey, zoneID),
-					resource.TestCheckResourceAttr(name, "flag", "false"),
-				),
+				Config: testLogpullRetentionSetConfig(rnd, zoneID, false),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						//plancheck.ExpectResourceAction(resourceName, plancheck.ResourceActionDestroy),
+						plancheck.ExpectKnownValue(resourceName, tfjsonpath.New("flag"), knownvalue.Bool(false)),
+					},
+				},
+				ConfigStateChecks: []statecheck.StateCheck{
+					statecheck.ExpectKnownValue(resourceName, tfjsonpath.New("flag"), knownvalue.Bool(false)),
+				},
+			},
+			// Set flag to true.
+			{
+				Config: testLogpullRetentionSetConfig(rnd, zoneID, true),
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						//plancheck.ExpectResourceAction(resourceName, plancheck.ResourceActionCreate),
```

Comment:
Commented out `plancheck.ExpectResourceAction()` for both steps, as I get error like below inconsistently:
```
go test ./internal/services/logpull_retention -run "^TestAccLogpullRetention" -v -count 1
=== RUN   TestAccLogpullRetention_Basic
    resource_test.go:28: Step 2/3 error: Pre-apply plan check(s) failed:
        'cloudflare_logpull_retention.mgvohhbmfy' - expected Create, got action(s): [delete create]
--- FAIL: TestAccLogpullRetention_Basic (7.11s)
FAIL
FAIL    github.com/cloudflare/terraform-provider-cloudflare/internal/services/logpull_retention 7.114s
FAIL
```

`logpull_retention` is based on a single flag / state per zone (and not multiple resources), so the state / action may be different based on initial state (before running the test).